### ISSUE TITLE
Publish-PsResource doesn't work on dev.azure.com repositories

### ISSUE THREAD

**Issue by Stephanevg** [PowerShell/PSResourceGet#1203]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

> **EDIT:  I have actually found the root of the issue.  Look at my second comment, I explain the details of the error.**

Hi all,

I am experiencing the following problem.

I am trying to publish a module to dev.azure.com feed.

I have created my self a test account, with full access to the feed.

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/bb40fa01-5c36-4fd5-a411-216ef937f230)

I go create a feed, and copy the feed url:

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/c4278e9e-1998-4221-bc5f-4733cdff9f22)

After struggling with getting this cmdlet to work (secretmanagement module needs to be installed, although it is not specified, nor installed as a prerequisite). I have used the following code to register the reposistory:

```powershell
$url = "https://pkgs.dev.azure.com/DigitalDistrict01/_packaging/Feed/nuget/v3/index.json"
$Token = "MyComplexSuPerDuperComplexToken123"
$userName = "User.MyName@Email.com"
$token_pscredential = ConvertTo-SecureString $Token -AsPlainText -Force
$Credentials = New-Object –TypeName System.Management.Automation.PSCredential –ArgumentList $userName, $token_pscredential


$PsCredentialInfo = $PsCredentialInfo = [Microsoft.PowerShell.PSResourceGet.UtilClasses.PSCredentialInfo]::new("SecretStore","feed_district01",$credentials)
Register-PSResourceRepository -Name "feed_district01" -Uri $Url -Force -Credential $PsCredentialInfo -Trusted -verbose
```

And it looks like I manage to register it successfully:

```powershell
get-psresourceRepository
```

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/031e4f63-2efb-4fc1-9be8-d91847d0e0c8)

I than try to publish a test module using the following command:

```

Publish-PSResource -Path "C:\Users\Stephane\Code\TestModule\" -Repository feed_district01 -Verbose -ApiKey (get-secret feed_district01)

```

And this is the result I get.

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/988ee847-9a1f-449d-8e70-2e8dfcde4bdd)

One other test I have done, is tried to change how the url looked, as I read that nuget v3 was not yet supported. 

So, the same method, but slightly changed my url from:

```powershell
$url = "https://pkgs.dev.azure.com/DigitalDistrict01/_packaging/Feed/nuget/v3/index.json"
```

to

```powershell
$url = "https://pkgs.dev.azure.com/DigitalDistrict01/_packaging/Feed/nuget/v2/"
```

Than I got the following message which a bit better, but quite not sufficient:

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/23188171-736f-4a7a-9e09-44fda53117c1)

I do not have the possibility to test this on other types of repositories (artifactory / github) so I don't know if this is something isolated only on dev.azure.com, or if I have maybe have forgotten something on the configuration side of things on Azure devops (yet, the account has full control...).

Looks forward for any feedback!

### Expected behavior

```console
I expect to be able to publish to a dev.azure.com repository.
```

### Actual behavior

```console
It throws.
```

### Error details

```console
PS C:\Users\Stephane\Code\TestModule> get-error

Exception             : 
    Type    : System.ArgumentException
    Message : Unable to load the service index for source https://pkgs.dev.azure.com/DigitalDistrict01/_packaging/Feed/nuget/v3/index.json.
    HResult : -2147024809
CategoryInfo          : InvalidResult: (:) [Publish-PSResource], ArgumentException
FullyQualifiedErrorId : PushNupkgError,Microsoft.PowerShell.PSResourceGet.Cmdlets.PublishPSResource
InvocationInfo        : 
    MyCommand        : Publish-PSResource
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 79
    Line             : Publish-PSResource -Path "C:\Users\Stephane\Code\TestModule\" -Repository feed_district01 -Verbose -ApiKey (get-secret feed_district01)
    PositionMessage  : At line:1 char:1
                       + Publish-PSResource -Path "C:\Users\Stephane\Code\TestModule\" -Reposi …
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Publish-PSResource
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : 


and

PS C:\Users\Stephane\Code\TestModule> get-error

Exception             : 
    Type    : System.ArgumentException
    Message : Repository 'feed_district01': Response status code does not indicate success: 401 (Unauthorized).
    HResult : -2147024809
CategoryInfo          : PermissionDenied: (:) [Publish-PSResource], ArgumentException
FullyQualifiedErrorId : 401Error,Microsoft.PowerShell.PSResourceGet.Cmdlets.PublishPSResource
InvocationInfo        : 
    MyCommand        : Publish-PSResource
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 88
    Line             : Publish-PSResource -Path "C:\Users\Stephane\Code\TestModule\" -Repository feed_district01 -Verbose -ApiKey (get-secret feed_district01)
    PositionMessage  : At line:1 char:1
                       + Publish-PSResource -Path "C:\Users\Stephane\Code\TestModule\" -Reposi …
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Publish-PSResource
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

### Environment data

```powershell
PS C:\Users\Stephane\Code\TestModule> get-module Microsoft.PowerShell.PSResourceGet;$PSVersionTable | ft

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     0.5.22     beta22     Microsoft.PowerShell.PSResourceGet  {Find-PSResource, Get-InstalledPSResource, Get-PSResourceRepository, Get-PSScriptFileInfo…}


Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by Stephanevg** [comment]
# intro

So, I actually found a way on how to get this to work.  Ultimatley, the Publish-PSRessource abstracts the original error message from nuget to much. Nuget actually gives valuable information. See below.

## troubleshooting steps

When I tried to upload to that exact feed using nuget, this is the message I got.

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/391a1189-b9c1-41f9-b9ec-590d613f61d5)

We see, that the module 'TestModule' already present is on one of the UpstreamSources.

My fix basically, was to rename my testing module to WoopModule.psd1 and WoopModule.psm1 ....

It than worked with Nuget.

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/0bb41c88-2721-42d3-9774-77601c3cad19)

And with Publish-PSResourceGet

![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/64f793f9-900d-40e4-981b-0d6b0306f736)

# To sum up

There are two issues which I belive should be looked at:
1) The parameter -Credential expects the wrong type
2) The nuget error message is abstracted, and actually gives valuable information back which is ignored by Publish-PSResource.

**Comment by SydneyhSmith** [comment]
We are investigating this issue but it seems to be an issue with how the credentials are being passed in the header to Azure Dev Ops, sorry about the issue

**Comment by Stephanevg** [comment]
FYI: I tried to remove and reregister the repository and I do have the same original problem now again. This time, my 'silly fix' does not help :(


![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/06fd322e-b3ee-41bd-87de-d32f45632fb9)

**Comment by Stephanevg** [comment]
hi @anamnavi 

How can I see when this release is published to the gallery? 
It is merged in master branch, but not available on the gallery yet.
![image](https://github.com/PowerShell/PSResourceGet/assets/6302961/6ff306f4-223e-498c-8ecd-dfd462225cfd)

thx
